client/driver/executor/executor.go

```diff
@@ -456,6 +481,36 @@ func (e *UniversalExecutor) DeregisterServices() error {
 	return nil
 }
 
+// PidStats returns the resource usage stats per pid
+func (e *UniversalExecutor) PidStats() (map[string]*cstructs.ResourceUsage, error) {
+	stats := make(map[string]*cstructs.ResourceUsage)
+	ts := time.Now()
+	for _, pid := range e.pids {
+		p, err := process.NewProcess(int32(pid.pid))
+		if err != nil {
+			e.logger.Printf("[DEBUG] executor: unable to create new process with pid: %v", pid.pid)
+			continue
+		}
+		ms := &cstructs.MemoryStats{}
+		if memInfo, err := p.MemoryInfo(); err == nil {
+			ms.RSS = memInfo.RSS
+			ms.Swap = memInfo.Swap
+		}
+
+		cs := &cstructs.CpuStats{}
+		if cpuStats, err := p.Times(); err == nil {
+			cs.SystemMode = cpuStats.System
+			cs.UserMode = cpuStats.User
+
+			// calculate cpu usage percent
+			cs.Percent = pid.cpuStats.Percent(cpuStats.Total())
+		}
+		stats[strconv.Itoa(pid.pid)] = &cstructs.ResourceUsage{MemoryStats: ms, CpuStats: cs, Timestamp: ts}
+	}
+
+	return stats, nil
```

Comment:
@dadgar I want to keep an `error` in the signature of `Stats`, just so that in the future it's easy for us to use that if we need.